demos/text_detection_demo/cpp_gapi/main.cpp

```diff
@@ -0,0 +1,509 @@
+                                            decoderStartIndex);
+                    } else if (decoderType == "ctc") {
+                        if (decoderBandwidth == 0) {
+                            res = CTCGreedyDecoder(outputData, trAlphabet, kPadSymbol, &conf);
+                        } else {
+                            res = CTCBeamSearchDecoder(outputData, trAlphabet, kPadSymbol, &conf,
+                                                       decoderBandwidth);
+                        }
+                    } else {
+                        slog::err << "No decoder type or invalid decoder type (-dt) provided: " <<
+                                     decoderType << slog::endl;
+                        return -1;
+                    }
+                    if (FLAGS_lower) {
+                        res = cv::toLowerCase(res);
+                    }
+                    res = conf >= trMinConfidence ? res : "";
+                    numRecognized += !res.empty() ? 1 : 0;
+                }
+
+                const auto &points = outPts[l];
+                if (rawOutput) {
+                    for (size_t i = 0; i < points.size(); i++) {
+                        slog::debug << clip(static_cast<int>(points[i].x), image.cols - 1) << "," <<
+                                       clip(static_cast<int>(points[i].y), image.rows - 1);
+                        if (i != points.size() - 1) {
+                            slog::debug << ",";
+                        }
+                    }
+                    if (trRequired) {
+                        slog::debug << "," << res;
+                    }
+                    if (!points.empty()) {
+                        slog::debug << slog::endl;
+                    }
+                }
+
+                // Displaying the results
+                if (!noShow && (!res.empty() || !trRequired || centralCrop)) {
+                    for (size_t i = 0; i < points.size() ; i++) {
+                        cv::line(demoImage, points[i], points[(i + 1) % points.size()],
+                                 cv::Scalar(50, 205, 50), 2);
+                    }
+                    if (!points.empty() && !res.empty()) {
+                        setLabel(demoImage, res, points[custom::getTopLeftPointIdx(points)]);
+                    }
+                }
+            }
+
+            // Displaying system parameters and FPS
+            putHighlightedText(demoImage, "Found: " + std::to_string(numRecognized),
+                               cv::Point(10, 80), cv::FONT_HERSHEY_COMPLEX, 0.65,
+                               cv::Scalar(0, 0, 255), 2);
+
+            presenter.drawGraphs(demoImage);
+            metrics.update(beginFrame, demoImage, { 10, 22 }, cv::FONT_HERSHEY_COMPLEX, 0.65);
+
+            if (videoWriter.isOpened() &&
+                (outputFramesLimit == 0 || metrics.getFrameCount() <= outputFramesLimit)) {
```

Comment:
It is implemented in the `ImagesCapture` class.

Reply:
You are about metrics, right? Yes, I didn't know, will change to use ImagesCapture metrics instead